holoviews/plotting/widgets/widgets.js

```diff
@@ -346,3 +337,198 @@ function update_widget(widget, values) {
     widget.trigger("change");
   };
 }
+
+function init_slider(id, plot_id, dim, values, next_vals, labels, dynamic, step, next_dim,
+                     dim_idx, delay, jQueryUI_CDN, UNDERSCORE_CDN) {
+	// Slider JS Block START
+  function loadcssfile(filename){
+    var fileref=document.createElement("link")
+    fileref.setAttribute("rel", "stylesheet")
+    fileref.setAttribute("type", "text/css")
+    fileref.setAttribute("href", filename)
+    document.getElementsByTagName("head")[0].appendChild(fileref)
+  }
+  loadcssfile("https://code.jquery.com/ui/1.10.4/themes/smoothness/jquery-ui.css");
+  /* Check if jQuery and jQueryUI have been loaded
+     otherwise load with require.js */
+  var jQuery = window.jQuery,
+    // check for old versions of jQuery
+    oldjQuery = jQuery && !!jQuery.fn.jquery.match(/^1\.[0-4](\.|$)/),
+    jquery_path = '',
+    paths = {},
+    noConflict;
+  var jQueryUI = jQuery.ui;
+  // check for jQuery
+  if (!jQuery || oldjQuery) {
+    // load if it's not available or doesn't meet min standards
+    paths.jQuery = jQuery;
+    noConflict = !!oldjQuery;
+  } else {
+    // register the current jQuery
+    define('jquery', [], function() { return jQuery; });
+  }
+  if (!jQueryUI) {
+	paths.jQueryUI = jQueryUI_CDN.slice(null, -3);
+  } else {
+    define('jQueryUI', [], function() { return jQuery.ui; });
+  }
```

Comment:
This is a hacky JQueryUI/regexp/require nightmare! I won't suggest touching it if it works but we should try to get rid if we ever see the opportunity!

Reply:
Yes, absolutely the first thing I want to get rid of. As I pointed out above, some of the widget code is salvageable. One of the first steps to doing that will be to get rid of the jQuery(UI) dependency. One of the reasons why I actually like a JupyterLab extension is because I don't have to fight require anymore.